osu.Game.Rulesets.Osu/Objects/Drawables/DrawableOsuBlinds.cs

```diff
@@ -0,0 +1,298 @@
+        {
+            this.restrictTo = restrictTo;
+
+            modEasy = hasEasy;
+            modHardrock = hasHardrock;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ISkinSource skin, TextureStore textures)
+        {
+            RelativeSizeAxes = Axes.Both;
+            Width = 1;
+            Height = 1;
+
+            Add(box1 = new Box
+            {
+                Anchor = Anchor.TopLeft,
+                Origin = Anchor.TopLeft,
+                Colour = Color4.Black,
+                RelativeSizeAxes = Axes.Y,
+                Width = 0,
+                Height = 1,
+                Depth = black_depth
+            });
+            Add(box2 = new Box
+            {
+                Anchor = Anchor.TopRight,
+                Origin = Anchor.TopRight,
+                Colour = Color4.Black,
+                RelativeSizeAxes = Axes.Y,
+                Width = 0,
+                Height = 1,
+                Depth = black_depth
+            });
+
+            Add(bgPanelLeft = new ModBlindsPanelSprite {
+                Origin = Anchor.TopRight,
+                Colour = Color4.Gray,
+                Depth = bg_panel_depth + 1
+            });
+            Add(panelLeft = new ModBlindsPanelSprite {
+                Origin = Anchor.TopRight,
+                Depth = bg_panel_depth
+            });
+
+            Add(bgPanelRight = new ModBlindsPanelSprite {
+                Origin = Anchor.TopLeft,
+                Colour = Color4.Gray,
+                Depth = fg_panel_depth + 1
+            });
+            Add(panelRight = new ModBlindsPanelSprite {
+                Origin = Anchor.TopLeft,
+                Depth = fg_panel_depth
+            });
+
+
+            random = new Random();
+            Add(bgRandomNpc = new Box
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Colour = Color4.Black,
+                Width = 512 * 0.4f,
+                Height = 512 * 0.95f,
+                RelativePositionAxes = Axes.Y,
+                X = -512,
+                Y = 0,
+                Depth = black_depth
+            });
+            Add(new SkinnableDrawable("Play/Catch/fruit-catcher-idle", name => randomNpc = new Sprite
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Texture = textures.Get(name),
+                Width = 512,
+                Height = 512,
+                RelativePositionAxes = Axes.Y,
+                X = -512,
+                Y = 0
+            }) {
+                Depth = npc_depth
+            });
+
+            this.skin = skin;
+            skin.SourceChanged += skinChanged;
+            PanelTexture = textures.Get("Play/osu/blinds-panel");
+        }
+
+        private void skinChanged()
+        {
+            PanelTexture = skin.GetTexture("Play/osu/blinds-panel");
+        }
+
+        private float applyGap(float value)
```

Comment:
A more readably version of this methis would be:

````csharp
private float applyGap(float value)
{
    const float easyMultiplier = 0.95f;
    const float hardrockMultiplier = 1.1f;

    float multiplier = 1;

    if (modEasy)
        multiplier = easyMultiplier;
    if (modHardrock)
        multiplier = hardrockMultiplier;

    float ret = value * multiplier;

    return Math.Clamp(ret, 0, targetClamp);
}
````

Reply:
MathHelper.Clamp then.

Please extend my example for adding CS and OD modifier.

example:

````csharp
private float applyGap(float value)
{
    const float easyMultiplier = 0.95f;
    const float hardrockMultiplier = 1.1f;

    float multiplier = 1;

    if (modEasy)
        multiplier = easyMultiplier;
    if (modHardrock)
        multiplier = hardrockMultiplier;
	
	multiplier *= (cs - 5) * 2;
	multiplier *= (od - 5) * 1.5;

    float ret = value * multiplier;

    return MathHelper.Clamp(ret, 0, targetClamp);
}
````

Reply:
It's MathHelper.Clamp IIRC